packages/lexical-markdown/src/MarkdownExport.ts

```diff
@@ -205,14 +205,23 @@ function exportTextFormat(
   // Where it would be invalid markdown to generate: "**   foo   **"
   // We instead want to trim the whitespace out, apply formatting, and then
   // bring the whitespace back. So our returned string looks like this: "   **foo**   "
+  // for whitespace only string case like: "  ", we replace the whitespace with &#160;
   const frozenString = textContent.trim();
   let output = frozenString;
+  const isFormattedWhitespace =
+    frozenString.length === 0 && node.getFormat() !== 0;
+
+  if (isFormattedWhitespace) {
+    // Convert whitespaces to the code entity
+    output = [...textContent]
+      .map((char) => '&#' + char.charCodeAt(0) + ';')
```

Comment:
Let's use fromCodePoint even though it doesn't make any difference for these specific whitespace characters we'll be encoding. HTML entities are based on code points and not UTF-16 code units. Might as well correctly support decoding any entity with this syntax.

```suggestion
      .map((char) => '&#' + char.codePointAt(0) + ';')
```